### DISCUSSION TITLE
Enabling the debug tap with only one ALT key?

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#12107] [Q&A] [answered]
I've read these instructions for enabling the debug tap.

>     Enable debug features by adding "debugFeatures": true, to the global settings
>     Hold both the left and right alt keys, and open a new tab. What you should see is a tab with two panes in it, and the second one will have the "debug tap", showing all the input and output sent to the Terminal.
> 

Two questions:

I have an old keyboard and RALT is key-mapped (registry) to RWIN.  So I don't have two Alt keys.  Do you think SendInput() will work in an EXE?

Are the instructions suggesting (for example, profile 1) the chord `LAlt + RAlt + Ctrl + Shift + 1`?  (Whew!)

Thanks!

**Comment by zadjii-msft** [comment]
I'm not sure it actually works at all with keybindings - the easier way is to just use the dropdown. Though, if you don't have two alt keys, then I dunno what to say. Temporarily remap it back to right alt? The debug tap isn't really a _robust_ scenario.

**Comment by vefatica** [comment]
Heehee!  I tried SendInput() ... LMENU down, RMENU down ... sleep a while while I do the dropdown ... RMENU up, LMENU up.

Not only did that not work, but it left the system thinking an Alt key was down.  I had to log out/in to fix that.  I wonder if the key-map (RALT -> RWIN) is triggered when I use SendInput().

**Comment by vefatica** [comment]
> The debug tap isn't really a robust scenario.

I'll say!  It works once in a while (I'd guess 1 in 8 tries).  But it usually just crashes windowsterminal.exe, leaving openconsole.exe (but usually not the shell) stranded.  Can I supply more than this?

```
Faulting application name: windowsterminal.exe, version: 1.11.2108.30001, time stamp: 0x612d15d9
Faulting module name: Microsoft.Terminal.Control.dll, version: 1.11.2108.30001, time stamp: 0x612d1456
Exception code: 0xc0000005
Fault offset: 0x00000000000107fe
```

**Comment by DHowett** [comment]
I suspect you're hitting #11282, which we also just found :smile:

**Comment by vefatica** [comment]
Sure does.  Sorry I missed that.

**Comment by j4james** [comment, marked as answer]
@vefatica One way you can trigger the debug tap without a second `Alt` key is by using the [on-screen keyboard](https://support.microsoft.com/en-us/windows/use-the-on-screen-keyboard-osk-to-type-ecbb5e08-5b4e-d8c8-f794-81dbf896267a). If you click on the `AltGr` key on that keyboard, it will temporarily lock it down as if it were pressed, and then you can press the `Alt` key on your physical keyboard (which will mean both of them are now pressed) while using the mouse to click the menu item for the profile you wish to launch.

**Comment by vefatica** [comment]
Thanks, @j4james.  That actually worked the first three times, and then windowsterminal crashed the next three times.  I had gotten around the difficulty by mapping RAlt to CapsLock.  That was easy since CapsLock was aleady in my keymap with no action (i.e., disabled).
